src/Microsoft.AspNet.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -26,26 +26,26 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
             try
             {
                 // Give application opportunity to find from a different location, adjust, or reject token
-                var messageReceivedContext = new MessageReceivedContext(Context, Options);
+                var receivingTokenContext = new ReceivingTokenContext(Context, Options);
 
                 // event can set the token
-                await Options.Events.MessageReceived(messageReceivedContext);
-                if (messageReceivedContext.HandledResponse)
+                await Options.Events.ReceivingToken(receivingTokenContext);
+                if (receivingTokenContext.HandledResponse)
                 {
-                    return messageReceivedContext.AuthenticationTicket;
+                    return receivingTokenContext.AuthenticationTicket;
                 }
 
-                if (messageReceivedContext.Skipped)
+                if (receivingTokenContext.Skipped)
                 {
                     return null;
                 }
 
                 // If application retrieved token from somewhere else, use that.
-                token = messageReceivedContext.Token;
+                token = receivingTokenContext.Token;
 
                 if (string.IsNullOrEmpty(token))
                 {
-                    string authorization = Request.Headers["Authorization"];
+                    var authorization = Request.Headers["Authorization"];
```

Comment:
If it's actually doing the challenge, then yeah, just `Challenge` I guess is fine.